core/src/test/java/org/jboss/jandex/test/BasicTestCase.java

```diff
@@ -475,39 +496,47 @@ public void testNullClass() throws IOException {
     }
 
     @Test
-    public void testRuntimeInvisiblePresentInV11() throws IOException {
-        @RuntimeInvisible(placement = "class")
-        class Target {
-            @SuppressWarnings("unused")
-            void execute(@RuntimeInvisible(placement = "arg") String arg) {
-            }
-        }
+    public void testRuntimeInvisiblePresentInV11() throws Exception {
+        // Check @RuntimeInvisible not available to reflection APIs
+        Class<RuntimeInvisibleTarget> testTarget = RuntimeInvisibleTarget.class;
+        assertEquals(0, testTarget.getDeclaredAnnotations().length);
+
+        Parameter[] executeParams = testTarget.getDeclaredMethod("execute", List.class).getParameters();
+        assertEquals(1, executeParams[0].getDeclaredAnnotations().length);
 
-        assertEquals(0, Target.class.getDeclaredAnnotations().length);
-        assertEquals(0, Target.class.getDeclaredMethods()[0].getDeclaredAnnotations().length);
+        AnnotatedParameterizedType apt = (AnnotatedParameterizedType) executeParams[0].getAnnotatedType();
+        AnnotatedType argGenericType = apt.getAnnotatedActualTypeArguments()[0];
+        assertEquals(1, argGenericType.getDeclaredAnnotations().length);
 
-        Index index = testClassConstantSerialisation(Index.of(Target.class), 11);
+        Index index = testClassConstantSerialisation(Index.of(testTarget), 11);
         DotName annoName = DotName.createSimple(RuntimeInvisible.class.getName());
         List<AnnotationInstance> rtInvisible = index.getAnnotations(annoName);
 
-        assertEquals(2, rtInvisible.size());
+        assertEquals(4, rtInvisible.size());
+        Map<String, List<AnnotationInstance>> placements = new HashMap<>(3);
 
         for (AnnotationInstance a : rtInvisible) {
             assertFalse(a.runtimeVisible());
+            String placement = a.value("placement").asString();
+            placements.compute(placement, (k, v) -> {
+                if (v == null) {
+                    v = new ArrayList<>();
+                }
+                v.add(a);
+                return v;
+            });
         }
+
+        assertEquals(1, placements.get("class").size());
+        // @RuntimeInvisible recorded on both method parameter and method parameter type
```

Comment:
This is interesting, but correct I guess? Since the annotation has `ElementType.PARAMETER, ElementType.TYPE_USE`.

Reply:
Yeah, exactly. I wasn't expecting it initially, but it makes sense.